plugins/modules/content_upload.py

```diff
@@ -113,11 +113,17 @@ def get_rpm_info(path):
     with open(path) as rpmfile:
         rpmhdr = ts.hdrFromFdno(rpmfile)
 
-    name = rpmhdr[rpm.RPMTAG_NAME].decode('ascii')
     epoch = rpmhdr[rpm.RPMTAG_EPOCHNUM]
-    version = rpmhdr[rpm.RPMTAG_VERSION].decode('ascii')
-    release = rpmhdr[rpm.RPMTAG_RELEASE].decode('ascii')
-    arch = rpmhdr[rpm.RPMTAG_ARCH].decode('ascii')
+    if isinstance(rpmhdr[rpm.RPMTAG_NAME], bytes):
+        name = rpmhdr[rpm.RPMTAG_NAME].decode('ascii')
+        version = rpmhdr[rpm.RPMTAG_VERSION].decode('ascii')
+        release = rpmhdr[rpm.RPMTAG_RELEASE].decode('ascii')
+        arch = rpmhdr[rpm.RPMTAG_ARCH].decode('ascii')
```

Comment:
Isn't there this strange `to_native` thing in ansible?

Reply:
There is, and I'm every time confused I when I have to use it…